be/src/services/vocabulary.service.ts

```diff
@@ -53,14 +57,21 @@ export class VocabularyService {
     // eslint-disable-rows-line @typescript-eslint/explicit-module-boundary-types
     public async listVocab(query: ListQueryModel, id?: string | ObjectId): Promise<ListingResult<Vocabulary>> {
         const {options, filter} = this.transformToMongoDBFilterOption(query, id);
-        // @ts-ignore
-        const dbos: Vocabulary[] = await this.collection.find(filter, options).toArray() as Vocabulary[];
-        return {
-            offset: query.offset,
-            rows: dbos.length,
-            totalItems: 0, // TODO
-            items: dbos
-        };
+        return Promise.all([
+            // @ts-ignore
+            this.collection.find(filter, options).toArray(),
+            this.countCollectionItems()
```

Comment:
This does not provide the correct number: It not about _all_ the vocabularies, but those that match the query/filter...